### PR TITLE
Allow `innerRef` prop to be passed down to composite components

### PR DESCRIPTION
**Pull request by Andarist**
Ref forwarding make the innerRef just a regular prop and this should be forwarded to the wrapped component

### CODE REVIEW

File: packages/styled-base/src/index.js [hunk 1]
```diff
@@ -142,11 +142,6 @@ let createStyled: CreateStyled = (tag: any, options?: StyledOptions) => {
               }
             } else {
               newProps.ref = ref
```
Comment (Andarist) [inline packages/styled-base/src/index.js]: `innerRef` is popular name in the react ecosystem and numerous libraries use it, IMHO it's undesirable to filter this out now as people might want to use it to get access to wrapped composite components

**Comment by Andarist** [comment]
We can keep a warning (although IMHO that is odd - even though I understand the concern that people might be confused). Anyhow I think we should forward this prop.

**Comment by mitchellhamilton** [comment]
I've been thinking about this a bit and I think we should maybe go so far as to still support innerRef for v10 and remove it in v11 since I think the migration experience is really important and there's no good way to safely codemod innerRef to ref.

**Comment by Andarist** [comment]
Personally I would be against supporting it - refs should be used with caution, so I wouldn't expect them being used in too many places per application.

I understand migration concerns though - it's always a hard choice between one and other, so my personal preference here (which boils down to not wanting to have redundant code & avoiding maintenance costs related to it) shouldn't matter as much.

I can adjust the PR in whatever way you see fit. 

Additional note - styled-components is dropping support for `innerRef` in v4, so at least it wouldn't be as unusual for us to drop it too.

**Comment by Andarist** [comment]
Closing in favour of https://github.com/emotion-js/emotion/pull/923

**Pull request closed without merging**